diff --git a/cmake/buildinfo_script.cmake b/cmake/buildinfo_script.cmake
index c7c68cd..d01a770 100644
--- a/cmake/buildinfo_script.cmake
+++ b/cmake/buildinfo_script.cmake
@@ -5,8 +5,8 @@ if (${GIT_FOUND} STREQUAL "TRUE")
     COMMAND ${GIT_EXECUTABLE} status --short --untracked=no --porcelain
     ERROR_VARIABLE GIT_STAT
     OUTPUT_VARIABLE GIT_OUTPUT
-    )
-  string(REGEX MATCH "not a git repo" NOT_GIT_VC "${GIT_STAT}")
+  )
+  string(REGEX MATCH ".*not a git repo.*" NOT_GIT_VC string(TOLOWER GIT_STAT "${GIT_STAT}"))
 else()
   set(NOT_GIT_VC -1)
 endif()
@@ -15,11 +15,13 @@ if (NOT_GIT_VC)
   #
   # Try to retrieve any saved git state from gitinfo.dat
   #
-  file(STRINGS "${CASTEP_SOURCE_DIR}/gitinfo.dat" GIT_SAVED_LIST)
-  foreach(GIT_LINE ${GIT_SAVED_LIST})
-    string(REGEX MATCH "([A-Za-z0-9_]+)=(.*)" GIT_LINE_MATCH ${GIT_LINE})
-    set(${CMAKE_MATCH_1} ${CMAKE_MATCH_2})
-  endforeach()
+  if (EXISTS "${CASTEP_SOURCE_DIR}/gitinfo.dat")
+    file(STRINGS "${CASTEP_SOURCE_DIR}/gitinfo.dat" GIT_SAVED_LIST)
+    foreach(GIT_LINE ${GIT_SAVED_LIST})
+      string(REGEX MATCH "([A-Za-z0-9_]+)=(.*)" GIT_LINE_MATCH ${GIT_LINE})
+      set(${CMAKE_MATCH_1} ${CMAKE_MATCH_2})
+    endforeach()
+  endif()
 else()
   execute_process(
     COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
